core/primitives/src/cpu.rs

```diff
@@ -0,0 +1,97 @@
+/// Try to ensure that the required CPU features are available and enabled for the current CPU.
+///
+/// Note that this is a best-effort kind of thing. If these features are already enabled when
+/// compiling the program, then we may very well end up executing some unsupported instruction
+/// before we even get to executing this code.
+pub fn ensure_cpu_compatibility(force: bool) {
+    if cfg!(near_production) || force {
+        imp::ensure_cpu_compatibility();
+    } else {
+        // Intentionally empty: allow execution without verifying the properties of the CPU.
```

Comment:
Perhaps it's better to `tracing::error` here?

Reply:
Or maybe return a `Result` out of this function and let the caller decide between log/process::exit/ignore ?